### ISSUE TITLE
[New cop] Check indentation of method chained multi-line block

### ISSUE THREAD

**Issue by pocke** [rubocop-hq/rubocop#5154]
I think there are three styles to write method chained multi-line block.

```ruby
# 1st - The indentation base is `.group_by`'s line.
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
  end

# 2nd - The indentation base for the block body is `.group_by`'s line,
#       and the indentation base for the `end` is `foobar`.
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
end

# 3rd - The indentation base is `foobar`.
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
  v.do_something(x)
end
```

Currently, RuboCop only warns 2nd style.

```bash
$ rubocop 
Inspecting 1 file
C

Offenses:

test.rb:13:1: C: Layout/IndentationWidth: Use 2 (not 4) spaces for indentation.
    v.do_something(x)
^^^^

1 file inspected, 1 offense detected
```

--------

## Expected behavior

I think 1st (or 2nd) style is better. So, If the others styles exist, a new cop should complain.
And maybe the styles should be configurable with `EnforcedStyle`.

## RuboCop version

master

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

Related issue (Japanese) https://github.com/rubocop-jp/issues/issues/25

**Comment by pocke** [comment]
And I have another style idea. (I'm sorry, I forgot this.)

```ruby
# 4th - assignment the `.group_by`'s result to avoid the indented block.
result = foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }

result.each do |v|
    v.do_something(x)
end
```

It is also good. And I think it is more readable than the 1st style.


> Question is if we should be relying on IndentationWidth here, or if there should be a MultiLineBlockBodyIndentation cop to take over. The former is already doing maybe a bit too much. WDYT?


I think we should add a new cop, because I want the two styles, like `assignment`(the 4th) and `indent`(the 1st).

**Comment by selesse** [comment]
+1 for a new cop.

Has there been any progress on this?

**Comment by bbatsov** [comment]
We should clearly aim to enforce 1. No progress yet - we're looking for a volunteer to tackle this one.

**Comment by hoshinotsuyoshi** [comment]
In 1st-3rd examples, I found that setting `Layout/BlockAlignment`  `EnforcedStyleAlignWith: start_of_block` may enforce 1st style. ⭐️ 

(Personally, I have not use this setting ever, but this seems so convenient that I will use it in my workplace project ... 😃 )

```yaml
Layout/BlockAlignment:
  EnforcedStyleAlignWith: start_of_block
```

----

## demo

### $ cat a.rb

```ruby
# 1st
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
  end

# 2nd
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
end

# 3rd
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
  v.do_something(x)
end
```

### $ cat .rubocop.yml

```yaml
Layout/BlockAlignment:
  EnforcedStyleAlignWith: start_of_block
```

### $ rubocop -a a.rb

```
Inspecting 1 file
C

Offenses:

a.rb:12:1: C: [Corrected] Layout/IndentationWidth: Use 2 (not 4) spaces for indentation.
    v.do_something(x)
^^^^
a.rb:12:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
  v.do_something(x)

a.rb:13:1: C: [Corrected] Layout/BlockAlignment: end at 13, 0 is not aligned with .group_by { something }.each do |v| at 11, 2.
end
^^^
a.rb:19:3: C: [Corrected] Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.
  v.do_something(x)

a.rb:20:1: C: [Corrected] Layout/BlockAlignment: end at 20, 0 is not aligned with .group_by { something }.each do |v| at 18, 2.
end
^^^

1 file inspected, 5 offenses detected, 5 offenses corrected
```


### $ cat a.rb

```ruby
# 1st
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
  end

# 2nd
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
  end

# 3rd
foobar
  .select { |v| something(v) }.reject { |v| something(v) }
  .group_by { something }.each do |v|
    v.do_something(x)
  end
```


### $ rubocop -V

```
0.68.0 (using Parser 2.6.3.0, running on ruby 2.5.1 x86_64-darwin17)
```
